src/Intervention/Image/Image.php

```diff
@@ -129,13 +129,18 @@ public function encode($format = null, $quality = 90)
     public function save($path = null, $quality = null)
     {
         $path = is_null($path) ? $this->basePath() : $path;
+        $dir = str_replace(strrchr($path, "/"), "", $path);
```

Comment:
This code would replace `/foo/bar.jpg/bar.jpg` (dumb, but possible) to `/foo`

I's suggest instead `$dir = dirname($path);`

Reply:
`$dir = dirname($path)` return `/foo/bar.jpg` but need to return `/foo`

Reply:
Thanks, but you should replace your line with mine, not put both ;)

Reply:
Another thing, it should be below the `if (is_null($path))` test ;)

Reply:
Being that `dirname($path)` return `/foo/bar.jpg` but need return `/foo`, the directory without the file at the end

Reply:
That's precisely what I intended, see:
```php
dirname('/foo/bar.jpg');         // '/foo'
dirname('/foo/bar.jpg/bar.jpg'); // '/foo/bar.jpg'
```
The 2nd example is dumb, but possible. My point was to demonstrate your code could encounter pitfalls.

Reply:
Don't you see that in my example there is a *directory* having the same name as the file?

Reply:
Why put the directory name with the same file name? No one does this!

Reply:
That's exactly what an edge case is.

This one demonstrates your code operates fundamentally wrong, even if it gives the expected result 99% of the time. If a code *can* fail, consider it *will*. See [Murphy's law](https://en.wikipedia.org/wiki/Murphy%27s_law).

And you certainly should not disregard a potential bug because "no one does this". Many people thinking like this ended up in [the daily wtf](https://thedailywtf.com/).

Reply:
Now it fails too often, in fact this is a library for developers and not for end users.

"99%" is "100%" in this case.

Reply:
Though I'd still rather recommend `dirname()`, a proper version of your code would be:
```php
$dir = substr($path, 0, strrpos($path, '/'));
```

Reply:
Yes, it gets better :)